returnn/search.py

```diff
@@ -14,6 +14,167 @@
+    ):
+        """
+        :param dict[str] search_data: dataset used for search
+        :param Checkpoint model_checkpoint:  TF model checkpoint. see `ReturnnTrainingJob`.
+        :param ReturnnConfig returnn_config: object representing RETURNN config
+        :param int log_verbosity: RETURNN log verbosity
+        :param str device: RETURNN device, cpu or gpu
+        :param float|int time_rqmt: job time requirement in hours
+        :param float|int mem_rqmt: job memory requirement in GB
+        :param float|int cpu_rqmt: job cpu requirement in GB
+        :param tk.Path|str|None returnn_python_exe: path to the RETURNN executable (python binary or launch script)
+        :param tk.Path|str|None returnn_root: path to the RETURNN src folder
+        """
+        self.search_data = search_data
+
+        self.model_checkpoint = model_checkpoint
+
+        self.returnn_python_exe = (
+            returnn_python_exe
+            if returnn_python_exe is not None
+            else gs.RETURNN_PYTHON_EXE
+        )
+
+        self.returnn_root = (
+            returnn_root if returnn_root is not None else gs.RETURNN_ROOT
+        )
+
+        self.out_returnn_config_file = self.output_path("returnn.config")
+
+        self.out_search_file = self.output_path("search_out")
+
+        self.returnn_config = self.create_returnn_config(
+            returnn_config, log_verbosity, device
+        )
+
+        self.rqmt = {
+            "gpu": 1 if device == "gpu" else 0,
+            "cpu": cpu_rqmt,
+            "mem": mem_rqmt,
+            "time": time_rqmt,
+        }
+
+    def tasks(self):
+        yield Task("create_files", mini_task=True)
+        yield Task("run", rqmt=self.rqmt)
+
+    def create_files(self):
+        config = self.returnn_config
+
+        config.config["load"] = self.model_checkpoint.ckpt_path
+        config.write(self.out_returnn_config_file.get_path())
+
+        cmd = [
+            tk.uncached_path(self.returnn_python_exe),
+            os.path.join(tk.uncached_path(self.returnn_root), "rnn.py"),
+            self.out_returnn_config_file.get_path(),
+        ]
+
+        util.create_executable("rnn.sh", cmd)
+
+        os.chmod(
+            "rnn.sh",
+            stat.S_IRUSR
+            | stat.S_IRGRP
+            | stat.S_IROTH
+            | stat.S_IWUSR
+            | stat.S_IXUSR
+            | stat.S_IXGRP
+            | stat.S_IXOTH,
+        )
+
+        # check here if model actually exists
+        assert os.path.exists(
+            tk.uncached_path(self.model_checkpoint.index_path)
+        ), "Provided model does not exists: %s" % str(self.model_checkpoint)
+
+    def run(self):
+        call = [
+            tk.uncached_path(self.returnn_python_exe),
+            os.path.join(tk.uncached_path(self.returnn_root), "rnn.py"),
+            self.out_returnn_config_file.get_path(),
+        ]
+        sp.check_call(call)
+
+    def create_returnn_config(self, returnn_config, log_verbosity, device):
```

Comment:
I was just wondering if we want to merge the `create_returnn_config` functions from the training and search job? But probably this could be a different issue/pull request

Reply:
what do you mean by merge? they are quite different right?

Reply:
I think discussing unifying "common" code can wait until we have all Returnn jobs pushed (e.g. I will also submit a "forward") job). As long as the behavior of the jobs does not change, refactoring can be done at any time.